velox/docs/develop/scalar-functions.rst

```diff
@@ -242,11 +242,41 @@ not go away prematurely. The output types can be scalar strings (varchar and
 varbinaries), but also complex types containing strings, such as arrays, maps,
 and rows.
 
+The :func:`setNoCopy` and :func:`setEmpty` methods of the out_type template
+can be used to set the result to a string in the input argument.
+
 .. code-block:: c++
 
   // Results refer to strings in the first argument.
   static constexpr int32_t reuse_strings_from_arg = 0;
 
+
+Here is an example of a zero-copy function:
+
+.. code-block:: c++
+
+  template <typename T>
+  struct UrlExtractProtocolFunction {
```

Comment:
Since the doc above talks about substr and trim, maybe use one of these functions for the example. You may want to simplify these implementation as opposed to copy as is.

Reply:
I've written a naive "Trim", would you mind take a look again?